### PR TITLE
Generalize `Mono.__eq__` to handle equality among subclasses.

### PR DESCRIPTION
**Pull request by kwmsmith**
Ensures `dshape('?string') == Option(string)` holds true.

ping @richafrank @llllllllll

### CODE REVIEW

File: datashape/coretypes.py [hunk 1]
```diff
@@ -87,7 +87,9 @@ def info(self):
         return type(self), self.parameters
 
     def __eq__(self, other):
-        return type(self) == type(other) and self.info() == other.info()
+        return (issubclass(type(other), Mono) and
```
Comment (richafrank) [inline datashape/coretypes.py]: Looks good to me.  I think `isinstance(other, Mono)` would be clearer.

File: datashape/coretypes.py [hunk 2]
```diff
@@ -87,7 +87,9 @@ def info(self):
...
+                self.shape == other.shape and
+                self.measure.info() == other.measure.info())
```
Comment (richafrank) [inline datashape/coretypes.py]: Does the `__hash__` need to be updated at all, or is `self.info()` still sufficient for equal-comparing instances to hash equal?

File: datashape/coretypes.py (hunk 1 above)
Comment (kwmsmith) [inline datashape/coretypes.py]: Yes, of course -- good catch, thanks.

File: datashape/coretypes.py (hunk 2 above)
Comment (kwmsmith) [inline datashape/coretypes.py]: Right again, good catch.

**Comment by richafrank** [comment]
👍 from me.  Great test coverage for these cases.

Thanks for the quick fix @kwmsmith .

**Pull request merged**